tdrs-backend/tdpservice/parsers/case_consistency_validator.py

```diff
@@ -0,0 +1,125 @@
+                    num_errors += self.__validate_ssp_s2_case(num_errors)
+                else:
+                    self.total_cases_validated -= 1
+                    logger.warn(f"Case: {self.current_case} has no errors but has either an incorrect program type: "
+                                f"{self.program_type} or an incorrect section: {self.section}. No validation occurred.")
+                    self.has_validated = False
+            else:
+                logger.debug(f"Case: {self.current_case} has errors associated with it's records. "
+                             "Skipping Cat4 validation")
+        return num_errors
+
+    def __validate_tanf_s1_case(self, num_errors):
+        """Perform TANF Section 1 category four validation on all cached records."""
+        num_errors += self.__validate_tanf_s1_header_with_records()
+        return num_errors
+
+    def __validate_tanf_s2_case(self, num_errors):
+        """Perform TANF Section 2 category four validation on all cached records."""
+        return 0
+
+    def __validate_tribal_tanf_s1_case(self, num_errors):
+        """Perform Tribal TANF Section 1 category four validation on all cached records."""
+        return 0
+
+    def __validate_tribal_tanf_s2_case(self, num_errors):
+        """Perform Tribal TANF Section 2 category four validation on all cached records."""
+        return 0
+
+    def __validate_ssp_s1_case(self, num_errors):
+        """Perform SSP Section 1 category four validation on all cached records."""
+        return 0
+
+    def __validate_ssp_s2_case(self, num_errors):
+        """Perform SSP Section 2 category four validation on all cached records."""
+        return 0
+
+    def __validate_tanf_s1_header_with_records(self):
+        """Header YEAR + header QUARTER must be consistent with RPT_MONTH_YEAR for all T1, T2, and T3 records."""
+        year = self.header["year"]
+        quarter = self.header["quarter"]
+        header_rpt_month_year_list = get_rpt_month_year_list(year, quarter)
+        num_errors = 0
+        for record, schema in self.record_schema_pairs:
+            if record.RPT_MONTH_YEAR not in header_rpt_month_year_list:
+                num_errors += 1
+                err_msg = (f"Failed to validate record with CASE_NUMBER={record.CASE_NUMBER} and "
+                           f"RPT_MONTH_YEAR={record.RPT_MONTH_YEAR} against header. If YEAR={year} and "
+                           f"QUARTER={quarter}, then RPT_MONTH_YEAR must be in {header_rpt_month_year_list}.")
+                self.generated_errors.append(
+                    self.generate_error(schema=schema,
+                                        error_category=ParserErrorCategoryChoices.CASE_CONSISTENCY,
+                                        error_message=err_msg,
+                                        record=record,
+                                        field="RPT_MONTH_YEAR"
+                                        )
```

Comment:
If there's some sort of failure here, could we consider wrapping in a try/catch and/or adding logging? If loop or conditional have issue, we'd just always return 0 or it might be difficult to debug why.

Reply:
I think we should move this comment to Jan's PR and have the `validate` method wrapped in a try catch if that makes sense to you?

Reply:
@andrew-jameson